src/parser.rs

```diff
@@ -554,7 +556,7 @@ impl<'a> Parser<'a> {
                 // issues here, we construct a new slice from raw parts, which
                 // then has lifetime bound to the outer function scope instead
                 // of the parser itself.
-                slice::from_raw_parts(self.buffer.as_ptr(), self.buffer.len())
+                slice::from_raw_parts(self.buffer[len..].as_ptr(), self.buffer.len() - len)
```

Comment:
fun fact, i forgot to add the `- len` here at the end at first and the next 3 bytes in memory on my machine were `red`, and it didn't segfault.